sql-plugin/src/main/scala/org/apache/spark/sql/rapids/HashFunctions.scala

```diff
@@ -44,6 +46,33 @@ case class GpuMd5(child: Expression)
   }
 }
 
+case class GpuSha1(child: Expression)
+  extends GpuUnaryExpression with ImplicitCastInputTypes with NullIntolerantShim {
+  override def toString: String = s"sha1($child)"
+  override def inputTypes: Seq[AbstractDataType] = Seq(BinaryType)
+  override def dataType: DataType = StringType
+
+  override def doColumnar(input: GpuColumnVector): ColumnVector = {
+    withResource(HashUtils.normalizeInput(input.getBase)) { normalized =>
+      // at the moment sha1 on CuDF doesn't support lists, so have to translate
+      // BinaryType into StringType first before we operate on it
+      val stringCV = withResource(normalized.getChildColumnView(0)) { dataCol =>
+        withResource(new ColumnView(DType.STRING, normalized.getRowCount,
+          Optional.of[java.lang.Long](normalized.getNullCount),
+          dataCol.getData, normalized.getValid, normalized.getOffsets)) { cv =>
+          cv.copyToColumnVector()
+        }
+      }
```

Comment:
Since we don't need `normalized` after this line, we can close it at line 65. Like the snippet below. Then use `normalizedStrCV` in the rest of the code.

```suggestion
    val normalizedStrCV = withResource(HashUtils.normalizeInput(input.getBase)) { normalized =>
      // at the moment sha1 on CuDF doesn't support lists, so have to translate
      // BinaryType into StringType first before we operate on it
      withResource(normalized.getChildColumnView(0)) { dataCol =>
        withResource(new ColumnView(DType.STRING, normalized.getRowCount,
          Optional.of[java.lang.Long](normalized.getNullCount),
          dataCol.getData, normalized.getValid, normalized.getOffsets)) { cv =>
          cv.copyToColumnVector()
        }
      }
    }
```

Reply:
Done